contracts/orbital-core/src/tests/unit_tests.rs

```diff
@@ -1,43 +1,166 @@
+#[should_panic(expected = "Error decoding bech32")]
+fn test_register_orbital_domain_validates_addr() {
+    let mut suite = Suite::default();
+
+    suite
+        .app
+        .execute_contract(
+            suite.owner,
+            suite.orbital,
+            &ExecuteMsg::RegisterNewDomain {
+                domain: "domain".to_string(),
+                account_type: AccountConfigType::Polytone {
+                    note: "invalid_note".to_string(),
+                    timeout: Uint64::one(),
+                },
+            },
+            &[],
+        )
+        .unwrap();
+}
+
+#[test]
+#[should_panic(expected = "Orbital domain already registered: ")]
+fn test_register_duplicate_orbital_domain() {
+    let mut suite = Suite::default();
+
+    suite
+        .app
+        .execute_contract(
+            suite.owner.clone(),
+            suite.orbital.clone(),
+            &ExecuteMsg::RegisterNewDomain {
+                domain: "".to_string(),
+                account_type: AccountConfigType::Polytone {
+                    note: suite.note.to_string(),
+                    timeout: Uint64::one(),
+                },
+            },
+            &[],
+        )
+        .unwrap();
+
+    suite
+        .app
+        .execute_contract(
+            suite.owner,
+            suite.orbital,
+            &ExecuteMsg::RegisterNewDomain {
+                domain: "".to_string(),
+                account_type: AccountConfigType::Polytone {
+                    note: suite.note.to_string(),
+                    timeout: Uint64::one(),
+                },
+            },
+            &[],
+        )
+        .unwrap();
+}
+
+#[test]
+#[should_panic(expected = "Caller is not the contract's current owner")]
+fn test_register_orbital_domain_validates_domain_owner() {
+    let mut suite = Suite::default();
+
+    suite
+        .app
+        .execute_contract(
+            suite.note.clone(),
+            suite.orbital,
+            &ExecuteMsg::RegisterNewDomain {
+                domain: "domain".to_string(),
+                account_type: AccountConfigType::Polytone {
+                    note: suite.note.to_string(),
+                    timeout: Uint64::one(),
+                },
             },
             &[],
-            "orbital-core",
-            None,
         )
         .unwrap();
+}
 
-    let resp: Ownership<String> = app
+#[test]
+#[should_panic(expected = "timeout must be non-zero")]
+fn test_register_orbital_domain_validates_timeout() {
+    let mut suite = Suite::default();
+
+    suite
+        .app
+        .execute_contract(
+            suite.owner,
+            suite.orbital,
+            &ExecuteMsg::RegisterNewDomain {
+                domain: "domain".to_string(),
+                account_type: AccountConfigType::Polytone {
+                    note: suite.note.to_string(),
+                    timeout: Uint64::zero(),
+                },
+            },
+            &[],
+        )
+        .unwrap();
+}
```

Comment:
ah interesting. I find them nice for very granular unit tests like checking some single parameter validation because they take up less lines of code and feel more concise.

@keyleu @stiiifff  any takes on this? happy to change to `unwrap_err()` if that's the consensus!

Reply:
I don't have a strong opinion about this. I generally use unwrap_err because then I compare the error directly with the error in error.rs so that if I change it there the test doesn't need to be changed. Here if you change the error string you need to also change what is in `expected`. Both are fine to me tbh

Reply:
I think `unwrap_err` is much cleaner, for example the comment above, I actually missed you were using the should_panic.
Adding to the reason keyne mentioned, having a code that panics is very weird, I think it is required for libraries that panic and you don't have a Result, but should not be done for contracts.

When you have a Result, its much cleaner to assert a specific error, because not only you can avoid changing the error string when you change it on the error type, but you can also assert dynamic errors with parameters easily.